web_advanced_search_x2x/static/src/js/web_advanced_search_x2x.js

```diff
@@ -0,0 +1,366 @@
+                }, '');
+                self.trigger('domain_selected', [['id', 'in', ids]], names);
+            });
+        },
+    });
+
+    instance.web.SearchView.include({
+        build_search_data: function()
+        {
+            //Advanced.commit_search can only cope with propositions
+            //(=domain leaves),
+            //so we need to rebuild the domain if one of our CompoundDomains
+            //is involved
+            var result = this._super.apply(this, arguments);
+            _.each(result.domains, function(domain, index)
+            {
+                if(!_.isArray(domain))
+                {
+                    return;
+                }
+                var compound_domains = [], leaves = [];
+                _.each(domain, function(leaf)
+                {
+                    if(leaf instanceof instance.web.CompoundDomain)
+                    {
+                        compound_domains.push(leaf);
+                    }
+                    if(_.isArray(leaf))
+                    {
+                        leaves.push(leaf);
+                    }
+                });
+                if(compound_domains.length)
+                {
+                    var combined = new instance.web.CompoundDomain();
+                    _.each(compound_domains, function(domain)
+                    {
+                        combined.add(domain.eval());
+                    })
+                    _.each(leaves, function(leaf)
+                    {
+                        combined.add([leaf])
+                    });
+                    result.domains[index] = combined;
+                }
+            });
+            return result;
+        },
+    })
+
+
+    instance.web.PermanentSearchView = instance.web.SearchView.include({
+        start: function()
+              {   var self = this;
+                  var p = this._super();
+                  instance.web.bus.on('click', this, function(ev) {
+                  //check if there is a class, avoid crashes
+                  if (( ev.target.attributes.class) &&  (ev.target.attributes))
+                      {
+                       curr_class = ev.target.attributes.class.value;
+                       //if the class is one of the search popups defined in this module, please don't close the search view when using them.
+                       if ((curr_class == "oe_button oe_selectcreatepopup-search-select oe_highlight") || 
+                           (curr_class == "oe_highlight oe_selectcreatepopup-search-select-domain")    ||
+                           (curr_class == "oe_button oe_selectcreatepopup-search-close oe_bold oe_form_button_cancel") ||
+                           (curr_class == "ui-dialog-titlebar-close ui-corner-all"))
+                           {
+                            if (this.$el)
+                                {
+                                 this.$el.addClass('oe_searchview_open_drawer');
```

Comment:
wouldn't is be a lot simpler to add those classes in `instance.web_advanced_search_x2x.SelectCreatePopup`?